File: Backend/admin_service/services/service_methodes.py
from sqlalchemy.orm import Session
from models.service import Service
from schemas.service_schema import ServiceCreate, ServiceUpdate

def create_service(db: Session, service: ServiceCreate) -> Service:
    new_service = Service(
        name=service.name,
        type=service.type,
        description=service.description
    )
    db.add(new_service)
    db.commit()
    db.refresh(new_service)
    return new_service


def update_service(db: Session, service_id: int, service_update: ServiceUpdate) -> Service | None:
    service = db.query(Service).filter(Service.id == service_id).first()
    if not service :
        return None
    
    if service_update.name is not None:
        service.name = service_update.name
    if service_update.type is not None:
        service.type = service_update.type
    if service_update.description is not None:
        service.description = service_update.description

    db.commit()
    db.refresh(service)
    return service


def delete_service(db: Session, service_id: int) -> bool:
    service = db.query(Service).filter(Service.id == service_id).first()

    if not service:
        return False
    
    db.delete(service)
    db.commit()
    return True


def get_services(db: Session) -> list[Service]:
    services = db.query(Service).all()
    return services

File: Frontend/Project_Front/lib/features/Admin/services/service_methodes.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

import 'package:authproject/config.dart' as config;
import '../models/service_model.dart';

class ServiceService {
  final String baseUrl = config.baseUrl;

  Future<List<ServiceModel>> getServices() async {
    final response = await http.get(Uri.parse("$baseUrl/service/services"));

    if (response.statusCode == 200) {
      final List data = jsonDecode(response.body);
      return data.map((e) => ServiceModel.fromJson(e)).toList();
    } else {
      throw Exception("Failed to load services");
    }
  }

  Future<ServiceModel> createService({
    required String name,
    required String type,
    String? description,
  }) async {
    final response = await http.post(
      Uri.parse("$baseUrl/service/create_service"),
      headers: {"Content-Type": "application/json"},
      body: jsonEncode({
        "name": name,
        "type": type,
        "description": description,
      }),
    );

    if (response.statusCode == 200 || response.statusCode == 201) {
      return ServiceModel.fromJson(jsonDecode(response.body));
    } else {
      throw Exception("Failed to create service");
    }
  }

  Future<ServiceModel> updateService(
    int id, {
    String? name,
    String? type,
    String? description,
  }) async {
    final response = await http.put(
      Uri.parse("$baseUrl/service/service_update/$id"),
      headers: {"Content-Type": "application/json"},
      body: jsonEncode({
        "name": name,
        "type": type,
        "description": description,
      }),
    );

    if (response.statusCode == 200) {
      return ServiceModel.fromJson(jsonDecode(response.body));
    } else {
      throw Exception("Failed to update service");
    }
  }

  Future<void> deleteService(int id) async {
    final response = await http.delete(Uri.parse("$baseUrl/service/$id"));

    if (response.statusCode != 200) {
      throw Exception("Failed to delete service");
    }
  }
}
